src/metabase/async/streaming_response.clj

```diff
@@ -81,10 +81,6 @@
       nil)
     (catch InterruptedException _
       (a/>!! canceled-chan ::thread-interrupted)
-      nil)
-    (catch Throwable e
-      (log/error e "Caught unexpected Exception in streaming response body")
-      (write-error! os e nil)
```

Comment:
Moved this handling of Throwable to `do-f-async` which is the only caller of this function. It already had a `catch Throwable` clause that couldn't ever happen.